src/main/java/org/sagebionetworks/bridge/config/PropertiesConfig.java

```diff
@@ -0,0 +1,243 @@
+
+    static final String USER_KEY = "bridge.user";
+    static final String ENV_KEY = "bridge.env";
+
+    private final String user;
+    private final Environment environment;
+    private final Properties properties;
+    private final Pattern delimiter;
+
+    /**
+     * @param configTemplate
+     *            Path to the config template file in the source code. The
+     *            config template should contain all the config entries. The
+     *            values can be default values, or, if sensitive, dummy values.
+     *            The values on the template file are to be overwritten by
+     *            environment variables and/or system properties in that order.
+     */
+    public PropertiesConfig(final Path configTemplate) throws IOException {
+        this(configTemplate, configTemplate, DEFAULT_LIST_DELIMITER);
+    }
+
+    /**
+     * @param configTemplate
+     *            Path to the config template file in the source code. The
+     *            config template should contain all the config entries. The
+     *            values can be default values, or, if sensitive, dummy values.
+     *            The values on the template file are to be overwritten by those
+     *            of the local config file, environment variables, and/or system
+     *            properties in that order.
+     *
+     * @param userConfig
+     *            Path to the default local config file which is inside the
+     *            user's home directory.
+     */
+    public PropertiesConfig(final Path configTemplate, final Path userConfig) throws IOException {
+        this(configTemplate, userConfig, DEFAULT_LIST_DELIMITER);
+    }
+
+    /**
+     * @param configTemplate
+     *            Path to the config template file in the source code. The
+     *            config template should contain all the config entries. The
+     *            values can be default values, or, if sensitive, dummy values.
+     *            The values on the template file are to be overwritten by those
+     *            of the local config file, environment variables, and/or system
+     *            properties in that order.
+     *
+     * @param userConfig
+     *            Path to the default local config file which is inside the
+     *            user's home directory.
+     *
+     * @param delimiterRegex
+     *            The regular expression for the delimiter that is used to
+     *            separate a list of values.
+     */
+    public PropertiesConfig(final Path configTemplate, final Path localConfig,
+            final String delimiterRegex) throws IOException {
+        checkNotNull(configTemplate);
+        checkNotNull(localConfig);
+        checkNotNull(delimiterRegex);
+        final Properties properties = new Properties();
+        try (final Reader templateReader = Files.newBufferedReader(configTemplate);
+                final Reader localReader = Files.newBufferedReader(localConfig)) {
```

Comment:
I've found that this doesn't work for Maven+Tomcat. In particular, a relative path is ambiguous (and an absolute path obviously doesn't work).

Best practices say you should use a classpath-relative path and use the class loader to load files. For example (where conf.conf is in the resources folder in the Maven project):
getClass().getClassLoader().getResourceAsStream("conf.conf")

If this doesn't work for Play or Gradle, the alternative is that the constructor takes an InputStream, so different consumers can either use Files or the class loader as appropriate.